src/System.Text.Json/src/System/Text/Json/Reader/JsonReaderOptions.cs

```diff
@@ -12,17 +12,34 @@ public struct JsonReaderOptions
         internal const int DefaultMaxDepth = 64;
 
         private int _maxDepth;
+        private JsonCommentHandling _commentHandling;
 
         /// <summary>
         /// Defines how the <see cref="Utf8JsonReader"/> should handle comments when reading through the JSON.
         /// By default <exception cref="JsonException"/> is thrown if a comment is encountered.
         /// </summary>
-        public JsonCommentHandling CommentHandling { get; set; }
+        /// <exception cref="ArgumentException">
+        /// Thrown when the comment handling enum is set to a value that is not supported (i.e. not within the <see cref="JsonCommentHandling"/> enum range).
+        /// </exception>
+        public JsonCommentHandling CommentHandling
+        {
+            get => _commentHandling;
+            set
+            {
+                if (!JsonHelpers.IsCommentHandlingSupported(value, JsonCommentHandling.Allow))
+                    throw ThrowHelper.GetArgumentException_CommentEnumMustBeInRange();
```

Comment:
Why do we use the `throw ThrowHelper.Get*` approach here rather than `ThrowHelper.Throw*`?